src/pages/Detail/Detail.js

```diff
@@ -1,7 +1,360 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import styled from 'styled-components';
+import Review from './Component/Review';
+import { FaStar } from 'react-icons/fa';
+import { FaHeart } from 'react-icons/fa';
+import { FaCartPlus } from 'react-icons/fa';
+import { FaArrowAltCircleDown } from 'react-icons/fa';
+import Recommend from './Component/Recommend';
 
 const Detail = () => {
-  return <div>This is DetailPage!!</div>;
+  const [bookData, setBookData] = useState([]);
+  const productParams = useParams();
+  const productId = productParams.id;
```

Comment:
구조분해할당해서 사용하실 수도 있습니다!
```suggestion
  const { id } = useParams();
```